### PR TITLE
Help Menu change

### PR DESCRIPTION
**Pull request by JamesDowney**
Ascii table is dumb, having to add things to ascii tables in lib.ts is dumb, I think this is better. The CLI is quite a pain to print to and like half of this PR is me reinventing the wheel with word wrap. Don't put < or > characters in the json unless you're trying to style the text I guess, printHtml doesn't play games.
image

### CODE REVIEW

File: src/lib.ts [hunk 1]
```diff
@@ -276,52 +277,25 @@ export function printLog(color: string): void {
...
-    +--------------------------+-----------------------------------------------------------------------------------------------+
-    |   garbo_autoUserConfirm  | **WARNING: Experimental** Don't show user confirm dialogs, instead automatically select yes/no|
-    |                          | in a way that will allow garbo to continue executing. Useful for scripting/headless. Risky and|
-    |                          |  potentially destructive.                                                                     |
-    +--------------------------+-----------------------------------------------------------------------------------------------+
-    |           Note:          | You can manually set these properties, but it's recommended that you use the relay interface. |
-    +--------------------------+-----------------------------------------------------------------------------------------------+</pre>`);
+  const helpData = Object.entries(JSON.parse(fileToBuffer("garbo_help.json")));
+  let rows = ``;
+  Object.entries(helpData).forEach(([, value]) => {
```
Comment (horrible-little-slime) [inline src/lib.ts:282]: wait, isn't helpData already obtained from an Object.entries()

Also, why use entries if you're going to ignore keys and only use values?

File: src/lib.ts [hunk 2]
```diff
@@ -276,52 +277,25 @@ export function printLog(color: string): void {
...
-    |       garbo_buyPass      | Set to true to buy a dinsey day pass with FunFunds at the end of the day, if possible.        |
-    +--------------------------+-----------------------------------------------------------------------------------------------+
-    |   garbo_autoUserConfirm  | **WARNING: Experimental** Don't show user confirm dialogs, instead automatically select yes/no|
-    |                          | in a way that will allow garbo to continue executing. Useful for scripting/headless. Risky and|
-    |                          |  potentially destructive.                                                                     |
-    +--------------------------+-----------------------------------------------------------------------------------------------+
-    |           Note:          | You can manually set these properties, but it's recommended that you use the relay interface. |
-    +--------------------------+-----------------------------------------------------------------------------------------------+</pre>`);
+  const helpData = Object.entries(JSON.parse(fileToBuffer("garbo_help.json")));
```
Comment (horrible-little-slime) [inline src/lib.ts:280]: Throw a type assertion on here to make your life easier; look at COMBO, for example

File: src/lib.ts [hunk 3]
```diff
@@ -276,52 +277,25 @@ export function printLog(color: string): void {
...
+    const val = [``];
+    valArray.forEach((element) => {
+      if (val[val.length - 1].length + element.length <= 82) {
+        val[val.length - 1] = `${val[val.length - 1]} ${element}`;
+      } else {
+        val[val.length - 1] = `${val[val.length - 1]}\n`;
+        val.push(element);
+      }
+    });
+    rows = `${rows}<tr><td width=200><pre> ${key}</pre></td><td width=600><pre>${
```
Comment (horrible-little-slime) [inline src/lib.ts:294]: did linting prevent you from using +=?

File: src/lib.ts [hunk 4]
```diff
@@ -276,52 +277,25 @@ export function printLog(color: string): void {
...
-    |           Note:          | You can manually set these properties, but it's recommended that you use the relay interface. |
-    +--------------------------+-----------------------------------------------------------------------------------------------+</pre>`);
+  const helpData = Object.entries(JSON.parse(fileToBuffer("garbo_help.json")));
+  let rows = ``;
+  Object.entries(helpData).forEach(([, value]) => {
+    const key = value[0];
+    const valArray = (value[1] as string).split(` `);
+    const val = [``];
+    valArray.forEach((element) => {
+      if (val[val.length - 1].length + element.length <= 82) {
```
Comment (horrible-little-slime) [inline src/lib.ts:287]: no magic numbers

**Review by horrible-little-slime — COMMENTED** [review]
I know I'm calling the kettle black here but I worry that this code may be hard to maintain because of legibility. I think it's in a situation where a little bit of work would go a long way--switching from `forEach` to `for`, being a little smarter about variable names, and using a type assertion on your JSON file might make it a little easier to follow

File: src/lib.ts (hunk 1 above)
Comment (gausie) [inline src/lib.ts:282]: Isn't this already entries?

File: src/lib.ts [hunk 5]
```diff
@@ -276,52 +277,25 @@ export function printLog(color: string): void {
...
-    +--------------------------+-----------------------------------------------------------------------------------------------+
-    |           Note:          | You can manually set these properties, but it's recommended that you use the relay interface. |
-    +--------------------------+-----------------------------------------------------------------------------------------------+</pre>`);
+  const helpData = Object.entries(JSON.parse(fileToBuffer("garbo_help.json")));
+  let rows = ``;
+  Object.entries(helpData).forEach(([, value]) => {
+    const key = value[0];
+    const valArray = (value[1] as string).split(` `);
+    const val = [``];
+    valArray.forEach((element) => {
```
Comment (gausie) [inline src/lib.ts:286]: Looks like this could instead be done with a flatmap if rhino supports it

File: src/lib.ts (hunk 1 above)
Comment (gausie) [reply src/lib.ts:282]: Also this probably wants to bea map or reduce or something

**Comment by gausie** [comment]
> switching from `forEach` to `for`

I'd be looking to go the other way and move to map and reduce.

File: src/lib.ts (hunk 5 above)
Comment (horrible-little-slime) [reply src/lib.ts:286]: iirc we have to polyfill it, but we do so in libram

File: src/lib.ts [hunk 6]
```diff
@@ -275,53 +276,32 @@ export function printLog(color: string): void {
...
+  const tableItems = helpData.map((x) => x.tableItem);
+  const descriptions = helpData.map((x) => x.description);
+  const updatedDescriptions: string[] = [];
+  descriptions.forEach((description) => {
+    if (description.length > tableMaxCharWidth) {
+      updatedDescriptions.push(description.replace(/(.{82}\s)/g, `$&\n`));
+    } else {
+      updatedDescriptions.push(description);
+    }
+  });
```
Comment (horrible-little-slime) [inline src/lib.ts:295]:
```suggestion
  const tableEntries = helpData.map((data) => ({
    description: data.description.length > tableMaxCharWidth ? data.description.replace(/(.{82}\s)/g, `$&\n`)) : data.description,
    tableItem: data.tableItem,
    }));
```

File: src/lib.ts [hunk 7]
```diff
@@ -275,53 +276,32 @@ export function printLog(color: string): void {
...
+  const rows = [];
+  for (let i = 0; i < helpData.length; i++) {
+    rows.push(
+      `<tr><td width=200><pre> ${tableItems[i]}</pre></td><td width=600><pre>${
+        updatedDescriptions[i] ?? ``
+      }</pre></td></tr>`
+    );
+  }
```
Comment (horrible-little-slime) [inline src/lib.ts]:
```suggestion
  const rows = tableEntries.map(({ tableItem, description }) => `<tr><td width=200><pre> ${tableItem}</pre></td><td width=600><pre>${description}</pre></td></tr>`);
  }
```

File: src/lib.ts (hunk 6 above)
Comment (horrible-little-slime) [reply src/lib.ts:295]: Wait, actually, just make the whole thing
```
  const rows = helpData.map((data) => {
    const choppedDescription =  data.description.length > tableMaxCharWidth ? data.description.replace(/(.{82}\s)/g, `$&\n`)) : data.description;
    return `<tr><td width=200><pre> ${data.tableItem}</pre></td><td width=600><pre>${choppedDescription}</pre></td></tr>`;
    ```

File: src/lib.ts [hunk 8]
```diff
@@ -275,53 +276,27 @@ export function printLog(color: string): void {
...
+    description:
+      data.description.length > tableMaxCharWidth
+        ? data.description.replace(/(.{82}\s)/g, `$&\n`)
+        : data.description,
+    tableItem: data.tableItem,
+  }));
+  const tableRows = tableEntries.map(
+    ({ tableItem, description }) =>
+      `<tr><td width=200><pre> ${tableItem}</pre></td><td width=600><pre>${description}</pre></td></tr>`
+  );
```
Comment (horrible-little-slime) [inline src/lib.ts:296]:
```suggestion
  const tableRows = helpData.map(({ tableItem, description }) => {
    const croppedDescription = data.description.length > tableMaxCharWidth
        ? data.description.replace(/(.{82}\s)/g, `$&\n`)
        : data.description;
    return `<tr><td width=200><pre> ${tableItem}</pre></td><td width=600><pre>${croppedDescription}</pre></td></tr>`;
  });
```

Comment (horrible-little-slime) [reply src/lib.ts:296]: just squashing both of those maps into one guy

**Review by horrible-little-slime — APPROVED** [review]

**Pull request merged**